File: js/programmers/level.0/prgm_120866_1.js
function solution(board) {
  const dx = [1, 0, -1, 0, 1, 1, -1, -1];
  const dy = [0, 1, 0, -1, 1, -1, 1, -1];
  const n = board.length;
  let result = n * n;
  const bomb = [];

  for (i = 0; i < n; i++) {
    for (j = 0; j < n; j++) {
      if (board[i][j] === 1) {
        bomb.push([i, j]);
        result--;
      }
    }
  }

  if (result === 0) return 0;

  bomb.forEach((v) => {
    for (k = 0; k < 8; k++) {
      const nx = v[0] + dx[k];
      const ny = v[1] + dy[k];

      if (nx >= 0 && ny >= 0 && nx < n && ny < n && board[nx][ny] === 0) {
        board[nx][ny] = 1;
        result--;
      }
    }
  });

  return result;
}

File: python/programmers/level.0/prgm_120866_1.py
def solution(board):
    n = len(board)
    dx = [-1, 1, 0, 0, -1, -1, 1, 1]
    dy = [0, 0, -1, 1, -1, 1, -1, 1]
    result = n * n
    bomb = []

    for i in range(n):
        for j in range(n):
            if board[i][j] == 1:
                bomb.append([i, j])
                result -= 1

    if result == 0: return 0

    for x, y in bomb:
        for i in range(8):
            nx = x + dx[i]
            ny = y + dy[i]

            if 0 <= nx < n and 0 <= ny < n and board[nx][ny] == 0:
                board[nx][ny] = 1
                result -= 1
    
    return result
